client/cli/go/cmds/load.go

```diff
@@ -0,0 +1,110 @@
+//
+// Copyright (c) 2015 The heketi Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cmds
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/heketi/heketi/apps/glusterfs"
+	client "github.com/heketi/heketi/client/api/go-client"
+	"github.com/spf13/cobra"
+)
+
+var jsonConfigFile string
+
+// Config file
+type ConfigFileNode struct {
+	Devices []string                 `json:"devices"`
+	Node    glusterfs.NodeAddRequest `json:"node"`
+}
+type ConfigFileCluster struct {
+	Nodes []ConfigFileNode `json:"nodes"`
+}
+type ConfigFile struct {
+	Clusters []ConfigFileCluster `json:"clusters"`
+}
+
+func init() {
+	RootCmd.AddCommand(loadCommand)
+
+	loadCommand.Flags().StringVarP(&jsonConfigFile, "json", "j", "",
+		"\n\tConfiguration containing devices, nodes, and clusters, in"+
+			"\n\tJSON format.")
+}
+
+var loadCommand = &cobra.Command{
+	Use:   "load",
+	Short: "Add devices to Heketi from a configuration file",
+	Long:  "Add devices to Heketi from a configuration file",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Check arguments
+		if jsonConfigFile == "" {
+			return errors.New("Missing configuration file")
+		}
+
+		// Load config file
+		fp, err := os.Open(jsonConfigFile)
+		if err != nil {
+			return errors.New("Unable to open config file")
+		}
+		defer fp.Close()
+
+		configParser := json.NewDecoder(fp)
+		var topology ConfigFile
+		if err = configParser.Decode(&topology); err != nil {
```

Comment:
Initializing and then the conditional! That's one pretty feature to have in a language. :+1: 
I would recommend you use this syntax in all the similar contexts! :-)